bob/src/cluster/quorum.rs

```diff
@@ -77,6 +78,38 @@ impl Quorum {
         }
     }
 
+    async fn delete_on_nodes(&self, key: BobKey, with_aliens: bool) -> Result<(), Error> {
+        debug!("DELETE[{}] ~~~DELETE LOCAL NODE FIRST~~~", key);
+        let mut local_put_ok = 0_usize;
+        let mut failed_nodes = Vec::new();
+        let res = delete_at_local_node(&self.backend, key, with_aliens).await;
```

Comment:
You should check, that it is required to delete from local node before doing so

Reply:
Are you sure it's needed here, and not on lower levels?

Reply:
This is needed to calculate `all_count` properly whithout call to `get_target_nodes_for_key` on this level.
Additionally, the function name by itself says, that it is supposed to work only for the local node. 
You could probably return an extra flag that tells whether it is required to run delete locally, but that would also require you to rename the function

Reply:
When `without_aliens` is true, we should check, that the current node contains target vdisk for key. Otherwise the error will be returned (https://github.com/qoollo/bob/blob/master/bob-backend/src/pearl/core.rs#L295).
But probably we should remove  `without_aliens` flag at all.